### ISSUE TITLE
bug: Error that causes the console to close automatically

### ISSUE THREAD

**Issue by EspectroW94** [AscensionGameDev/Intersect-Engine#1977]
### Description

I don't know exactly what the error is, or how to reproduce it. but I know what happens right after completing a mission. I was using version 0.8.0.14 when I realized the error. I updated to 0.8.0.18 to confirm if the error was fixed. but it still exists.

I leave the console log:
--------------------------------------------------------------------------------
```
2023-10-16 23:49:10.635 [Error] Jackpot! Concurrency Bug For EspectroW in Created Context 
2023-10-16 23:49:10.635 [Error] DbUpdateConcurrencyException: The database operation was expected to affect 1 row(s), but actually affected 0 row(s); data may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=527962 for information on understanding and handling optimistic concurrency exceptions.
    Stack:    at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ThrowAggregateUpdateConcurrencyException(RelationalDataReader reader, Int32 commandIndex, Int32 expectedRowsAffected, Int32 rowsAffected)
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeResultSetWithRowsAffectedOnly(Int32 commandIndex, RelationalDataReader reader)
   at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.Consume(RelationalDataReader reader)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(StateManager stateManager, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.<>c.<SaveChanges>b__107_0(DbContext _, ValueTuple`2 t)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Intersect.Server.Database.IntersectDbContext`1.SaveChanges(Boolean acceptAllChangesOnSuccess) in D:\a\Intersect-Engine\Intersect-Engine\Intersect.Server\Database\IntersectDbContext.cs:line 171
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at Intersect.Server.Database.IntersectDbContext`1.SaveChanges() in D:\a\Intersect-Engine\Intersect-Engine\Intersect.Server\Database\IntersectDbContext.cs:line 163
   at Intersect.Server.Database.PlayerData.User.Save(PlayerContext playerContext, Boolean force, Boolean create) in D:\a\Intersect-Engine\Intersect-Engine\Intersect.Server\Database\PlayerData\User.cs:line 375

--------------------------------------------------------------------------------
2023-10-16 23:49:10.637 [Error] Entry Type [Microsoft.EntityFrameworkCore.ChangeTracking.EntityEntry / Modified]
--------------------
Type: Intersect.Server.Database.PlayerData.Players.Quest
	Id (Token: False): Proposed: 3d03eb06-98dc-4477-baed-5ed2a38c1c3c  Original Value: 3d03eb06-98dc-4477-baed-5ed2a38c1c3c  Database Value: null
	Completed (Token: False): Proposed: False  Original Value: False  Database Value: null
	PlayerId (Token: False): Proposed: 7b96997a-fd50-432e-adb0-4a71772cf619  Original Value: 00000000-0000-0000-0000-000000000000  Database Value: null
	QuestId (Token: False): Proposed: 8464200d-299f-41ab-a410-03b960af831a  Original Value: 8464200d-299f-41ab-a410-03b960af831a  Database Value: null
	TaskId (Token: False): Proposed: 6faa1f2f-db2d-4344-814c-945a2cd3d46a  Original Value: 6faa1f2f-db2d-4344-814c-945a2cd3d46a  Database Value: null
	TaskProgress (Token: False): Proposed: 0  Original Value: 0  Database Value: null




--------------------------------------------------------------------------------
2023-10-16 23:49:10.639 [Error] Received unhandled exception from System.Threading.Thread: Failed to save user, shutting down to prevent rollbacks!

--------------------------------------------------------------------------------
2023-10-16 23:49:10.639 [Error] Exception: Failed to save user, shutting down to prevent rollbacks!
```

### Steps to Reproduce

I'm not sure

### Version with bug

0.8.0.18

### Last version that worked well

0.7.2.64

### Affected platforms

Windows

### Duplicate Bug Check

- [X] This bug report is not a duplicate to the best of my knowledge.

**Comment by EspectroW94** [comment]
I'm not sure, but I think this happens only by creating new users and completing some quests.

**Comment by lodicolo** [comment]
Fixed by 4d1b70d526c5a164cf518aa520eba00b083312d0
